### DISCUSSION TITLE
Pistons optimization

### DISCUSSION THREAD

**Discussion by RVSkeLe** [PurpurMC/Purpur#1505] [Feature Requests]
Hello, I'm currently trying to optimize all types of farms which use redstone mechanisms, tbe redstone dust and hoppers are already optimized, but other blocks such as pistons and droppers still give a little struggle to server performance.
It would be great if it was possible to optimize pistons too, since some farms running with ~400 pistons are taking ~2ms (as per Spark).

**Comment by 1ByteBit** [comment]
Isn't that already extremely optimized? Also you should suggest performance improvements (of course, with an idea of how to improve said performance instead of requesting better performance) to paper since purpur doesn't actually do anything performance wise
